ios_sample_test_app/ios_sample_test_app/AppDelegate.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  AppDelegate.swift
+//  ios_sample_test_app
+//
+//  Created by Brandon Boothe on 3/23/23.
+//
+
+import UIKit
+
+import BranchSDK
+
+@main 
+class AppDelegate: UIResponder, UIApplicationDelegate {
+    var window: UIWindow?
+    var navigationcontroller: UINavigationController?
+    
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        // Override point for customization after application launch.
+        
+        //See Branch Logs
+        Branch.getInstance().enableLogging()
+        
+        // This version of initSession includes the source UIScene in the callback
+        BranchScene.shared().initSession(launchOptions: launchOptions, registerDeepLinkHandler: { (params, error, scene) in
+            
+            guard let data = params as? [String: AnyObject] else { return }
+            
+            guard let options = data["nav_to"] as? String else { return }
+            switch options {
+                case "color_block_page":
+                
+                    let rootViewController = UIApplication.shared.windows.first?.rootViewController as? UINavigationController
+                    let secondVC = ColorBlockViewController()
+                    rootViewController?.pushViewController(secondVC, animated: true)
+                
+                default: break
+              }
+        })
+        
+        // login
+        Branch.getInstance().setIdentity("your_user_id")
+
+        // logout
+        Branch.getInstance().logout() // or .logoutWithCallback() to customize further
```

Comment:
Is the logout immediately after setting identity intentional?

Reply:
I had place it there just following [the Branch Docs here.](https://help.branch.io/developers-hub/docs/ios-basic-integration#track-users). I can comment it out or remove it though! Thoughts?

Reply:
Depends on your intent. The docs have it like that, I presume, to demonstrate the calls, not a real implementation. The identity is set on the session and any calls thenceforth will have that id. 

But, `logout` removes that a few milliseconds later. Maybe you want to call these functions behind buttons?

Reply:
Ok that makes sense. I removed it from AppDelegate and placed it behind the buttons that send events.

Reply:
Thank you